### DISCUSSION TITLE
When will `LEGO_EXPERIMENTAL_CNAME_SUPPORT` no longer be experimental?

### DISCUSSION THREAD

**Discussion by danthegoodman1** [go-acme/lego#1676] [Q&A] [answered]
It seems that it's been in the experimental stage for ~2 years. Is this feature being considered for full support?

Diving into the code, it seems simple and robust enough for it to be fully supported, maybe beyond the lack of an error check when reading the env var (but I guess that would not be needed in full support).

Is there somethingt that blocks this from being in full support now?

**Comment by ser** [comment]
Yeah, I think after 5 years it could be said it's not experimental anymore. It took me a while to find its existence.

**Reply by ldez** [reply, marked as answer]
This env var `LEGO_EXPERIMENTAL_CNAME_SUPPORT` was removed 2 years ago (2022/09/19).

Now, the CNAME support is by default, and you have an env var (`LEGO_DISABLE_CNAME_SUPPORT`) to disable it.

**Comment by ser** [comment]
Where is it documented? Here is nothing: https://go-acme.github.io/lego/dns/rfc2136/

**Reply by ldez** [reply]
This page is the page about a provider.

What are you looking for exactly? An option to enable CNAME or disable it?

The env var (`LEGO_DISABLE_CNAME_SUPPORT`) to disable CNAME was introduced only for internal debugging purposes, so it's not documented.

The support of CNAME is documented here:
- https://go-acme.github.io/lego/
- https://go-acme.github.io/lego/usage/cli/options/#dns-resolvers-and-challenge-verification
